skglm/solvers/lbfgs.py

```diff
@@ -40,13 +41,20 @@ def objective_function(w):
 
             return datafit_value + penalty_value
 
-        def jacobian_function(w):
+        def d_jacobian_function(w):
```

Comment:
small discussion:
- I think `_function` can be removed 
- the jacobian is a gradient here because the output dimension is 1

If we say jacobian to match scipy (which is perfectly fine), I think we can go with `jacobian` or `jac` (+ the s/d prefix)